### PR TITLE
added custom messages in context

### PR DESCRIPTION
**Pull request by ggofron**

### CODE REVIEW

File: src/base/Validation.tsx [hunk 1]
```diff
@@ -22,4 +22,14 @@ export class Validation<T> {
                 "Please type phone in format +[country prefix (1 to 3 digits)][local phone number (9 digits)]")
         )
     }
+    public static requiredInputCustomValidation(i18Wrapper: IntlWrapper, pattern, message): Validation<string> {
```
Comment (pdolega) [inline src/base/Validation.tsx]: what's the type of `pattern` ?

File: src/base/Validation.tsx [hunk 2]
```diff
@@ -22,4 +22,14 @@ export class Validation<T> {
...
+        let _pattern = pattern.split('/'),
+            finalPattern = _pattern[1];
```
Comment (pdolega) [inline src/base/Validation.tsx]: why do we split _pattern ?

File: src/base/Validation.tsx [hunk 3]
```diff
@@ -22,4 +22,14 @@ export class Validation<T> {
...
+        let inputRegEx = new RegExp(finalPattern);
+
+        return new Validation<string>(
+            text => text.match(inputRegEx),
+            text => i18Wrapper.format(message.id, message.text)
```
Comment (pdolega) [inline src/base/Validation.tsx]: is it possible that i18Wrapper is null ?

Comment (ggofron) [inline src/base/Validation.tsx]: it's gonna return default message

**Comment by pdolega** [comment]
Multiselect validation stopped working - except that works fine.

image

**Comment by ggofron** [comment]
target branch is wrong, ! created new PR with correct target branch

**Pull request closed without merging**